backend/src/models/service.ts

```diff
@@ -219,14 +221,16 @@ export class Service extends BaseEntity {
       }
       const product = {
         name: this.censysMetadata.product,
-        version: this.censysMetadata.version,
+        version: this.censysMetadata.version || undefined,
         description: this.censysMetadata.description,
         product: this.censysMetadata.product,
         revision: this.censysMetadata.revision,
         cpe,
         tags: []
       };
-      if (cpe) products[cpe] = product;
+      if (product.cpe && products[product.cpe])
```

Comment:
Can you add a test case for this? That test case would then document exactly what situations this overwriting logic is meant to be used for.

Reply:
Added tests, and fixed the logic a bit